File: leetcode/remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        prev = nums[0] - 1
        count = 0
        for x in range(len(nums)):
            if nums[x] > prev:
                nums[count] = nums[x]
                prev = nums[count]
                count += 1
        return count

File: leetcode/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int cnt=0;
        int prev = nums[0]-1;
        int n = nums.size();
        for(int x = 0;x<n;++x){
            if(nums[x]>prev){
                nums[cnt]=nums[x];
                prev = nums[cnt];
                cnt++;
            }
        }
        return cnt;
    }
};